src/main/java/alice/storage/TaskList.java

```diff
@@ -0,0 +1,121 @@
+package alice.storage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import alice.task.InvalidTaskException;
+import alice.task.Task;
+
+/** Container for tasks. */
+public class TaskList {
+    private final Storage storage;
+    private List<Task> tasks;
+
+    /**
+     * Creates a list to manage tasks.
+     */
+    public TaskList(Storage storage) {
+        this.storage = storage;
+        try {
+            this.tasks = loadTasks();
+        } catch (IOException | InvalidTaskException e) {
+            this.tasks = new ArrayList<>();
+        }
+    }
+
+    /**
+     * Basic getter for tasks.
+     *
+     * @return the current list of tasks
+     */
+    public List<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Adds task to list and saves to disk.
+     *
+     * @param  task        the task to add
+     * @throws IOException if the tasks cannot be saved
+     */
+    public void addTask(Task task) throws IOException {
+        tasks.add(task);
+        saveTasks();
```

Comment:
I understand that saveTasks() is called whenever a task is modified (added, deleted, marked). Is there any reason why saveTasks is not called at the end, when user says goodbye?